### ISSUE TITLE
Getting corresponding class names for the dataset

### ISSUE THREAD

**Issue by bhattbhuwan13** [facebookresearch/vissl#341]
## ❓ How do I find the class-label mapping for my own dataset

I wanted to fine-tune the resnet model to classify the x-ray image as either **normal or pneumonia**. For the same, I used the configuration file [eval_resnet_8gpu_transfer_in1k_fulltune.yaml](https://dl.fbaipublicfiles.com/vissl/tutorials/configs/eval_resnet_8gpu_transfer_in1k_fulltune.yaml) to train the model on single gpu computer. 

I trained the model for 7 epochs and the performance Is acceptable, I get the following results on test set:  
{'top_1': {0: 68.75}, 'top_2': {0: 100.0}} 

## I have two questions:
1. How do I find the class to label mapping
2. In the metrics, {'top_1': {0: 68.75}, 'top_2': {0: 100.0}} , what does the index 0 represent?

**Comment by prigoyal** [comment]
Hi @bhattbhuwan13 , thank you for reaching out about this. For the ImageNet, in the above config file, we are basically using the PyTorch `ImageFolder` https://github.com/facebookresearch/vissl/blob/master/vissl/data/disk_dataset.py#L85. To get the class information, one can use the `image_dataset.class_to_idx` where `class_to_idx` etc are exposed by pytorch https://github.com/pytorch/vision/blob/562b8463bcce05a958023cfe9c2e72185cb804d8/torchvision/datasets/folder.py#L152

**Comment by QuentinDuval** [comment]
Hi @bhattbhuwan13,

Regarding the second part of your question and the meaning of `{'top_1': {0: 68.75}, 'top_2': {0: 100.0}}`:

- The `top_1` and `top_2` correspond to the top 1 accuracy and top 2 accuracy
- The `0` as key is supposed to refer to the name of the layer where the accuracy is evaluated (this format is most useful for our configurations in which we do linear evaluation on multiple layers at once and need to give a name to each output)

The fact that `top_2` is there indicates that you probably changed some options in the configuration of the `eval_resnet_8gpu_transfer_in1k_fulltune.yaml`, am I correct? If not, there is an issue on our side.

Since in your case there are but 2 categories (normal or pneumonia), only the `top_1` accuracy makes sense. You can only output this one by modifying the configuration like so:

```
  METERS:
    name: accuracy_list_meter
    accuracy_list_meter:
      num_meters: 1
      topk_values: [1] # get rid of the top-5 or top-2
```

**Comment by bhattbhuwan13** [comment]
@QuentinDuval , thanks for a concise response. Yes, I modified the `topk_values` parameter. And it makes sense to get 100% accuracy `top_2` because there are only 2 unique classes in the dataset. Thanks.
